src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Helpers.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Diagnostics
+{
+    internal static class Helpers
+    {
+        internal static string FormatTags(IEnumerable<KeyValuePair<string, object?>>? tags)
+        {
+            if (tags is null)
+            {
+                return string.Empty;
+            }
+
+            StringBuilder sb = new StringBuilder();
+            bool first = true;
+            foreach (KeyValuePair<string, object?> tag in tags)
+            {
+                if (first)
+                {
+                    first = false;
+                }
+                else
+                {
+                    sb.Append(',');
+                }
+
+                sb.Append(tag.Key).Append('=');
+
+                if (tag.Value is not null)
+                {
+                    sb.Append(tag.Value.ToString());
+                }
+            }
+            return sb.ToString();
+        }
+
+        internal static string FormatTags(KeyValuePair<string, string>[] labels)
+        {
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < labels.Length; i++)
+            {
+                sb.Append(labels[i].Key).Append('=').Append(labels[i].Value);
+                if (i != labels.Length - 1)
+                {
+                    sb.Append(',');
+                }
+            }
+            return sb.ToString();
+        }
+
+        internal static string FormatObjectHash(object? obj) =>
+            obj is null ? string.Empty : RuntimeHelpers.GetHashCode(obj).ToString(CultureInfo.InvariantCulture);
```

Comment:
Are you explicitly trying to avoid any object.GetHashCode override the object might provide?

Reply:
I want to generate instance hash code. we can pass any object (like string or any other type objects) and we want to differentiate between two instances that can generate same hash code  using `object.GetHashCode`.

Reply:
> and we want to differentiate between two instances that can generate same hash code using object.GetHashCode

RuntimeHelpers.GetHashCode(obj) does not guarantee that two distinct objects will produce different hash codes.

Reply:
Yes, but it is good enough for the cases we are using with Meter and Instruments for instance. The only case is the scope if someone created instances of types generating same hash. So `RuntimeHelpers.GetHashCode` is still better to use than `object.GetHashCode`. Note, we are trying to help in every corner case here which is someone create instrument/Meter with exact same parameters as names, tags...etc. If you have a better suggestion, please let me know and I can consider it.

Reply:
I updated the code and avoided sending the hash codes of the instrument and meter, instead I am sending a unique generated instrument id which help in teh scenarios we are trying fix.